jwst/pipeline/calwebb_coron3.py

```diff
@@ -46,17 +46,25 @@ class Coron3Pipeline(Pipeline):
 
     prefetch_references = False
 
+    # Main processing
     def process(self, user_input):
-        """Primary method for performing pipeline."""
+        """Primary method for performing pipeline.
+
+        Parameters
+        ----------
+        user_input: str, Level3 Association, or ~jwst.datamodels.DataModel
+            The exposure or association of exposures to process
+        """
         self.log.info('Starting calwebb_coron3 ...')
+        asn_exptypes = ['science', 'psf']
 
-        # Load the input association table
-        asn = self.load_as_level3_asn(user_input)
-        acid = asn.get('asn_id', '')
+        # Create a DM object using the association table
+        input_models = datamodels.open(user_input, asn_exptypes=asn_exptypes)
+        acid = input_models.meta.__dict__['_instance']['asn_table']['asn_id']
```

Comment:
To avoid the `__dict__`, should be able to do something similar as in line 66 below

`acid = input_models.meta.asn_table.asn_id`